community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/schema/LuceneSchemaIndex.java

```diff
@@ -97,6 +97,7 @@ public IndexReader getIndexReader() throws IOException
     public void verifyUniqueness( PropertyAccessor accessor, int propertyKeyId )
             throws IOException, IndexEntryConflictException
     {
+        flush( true );
```

Comment:
It turns out that this was a problem previously too, it just happened very rarely. If the index wasn't flushed and merged properly then the uniqueness check, which checks term frequencies on all "leaves" would sometimes see multiple leaves for the same key and hence miss non-uniqueness. Merging all segments together has lucene report correct counts.